src/main/groovy/SDKTester.groovy

```diff
@@ -277,20 +317,22 @@ class SDKTester {
 
             assert attach(s, v)
             println "attached $v to $s"
-            assert attachedVolumes(s).contains(v.id)
+            assert attachedVolumes(s).collect { it.id }.contains(v.id)
+            assert attachedVolume(s, v).id.equals(v.id)
 
             assert detach(s, v)
             println "detached $v from $s"
 
             assert attach(s, i)
             println "attached $i to $s"
-            assert attachedCDROMs(s).contains(i.id)
+            assert attachedCDROMs(s).collect { it.id }.contains(i.id)
```

Comment:
The `Image` object return from the `attachedCDROMs` differs from the `i` object in some properties. To simplify the assertion comparing id's is sufficient.